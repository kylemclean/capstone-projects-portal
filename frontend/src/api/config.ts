import { AxiosRequestConfig } from "axios"

// development config
const devConfig: AxiosRequestConfig = {
    baseURL: process.env.REACT_APP_API_BASE_URL ?? "http://localhost:8000/api",
    timeout: 3000,
}

// production config
const prodConfig: AxiosRequestConfig = {
    baseURL: process.env.REACT_APP_API_BASE_URL ?? "/api",
    timeout: 3000,
}

export default process.env.NODE_ENV === "production" ? prodConfig : devConfig

export const githubAuthorizeUrl =
    process.env.REACT_APP_GITHUB_AUTHORIZE_URL ??
    "https://github.com/login/oauth/authorize"
