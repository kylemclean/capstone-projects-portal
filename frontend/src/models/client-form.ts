export default interface ClientForm {
    name: string
    about?: string
    image?: string | BlobPart | null
    website_link?: string
    testimonial?: string
}
