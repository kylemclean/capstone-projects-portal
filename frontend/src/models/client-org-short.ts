import ClientOrgType from "./client-org-type"

export default interface ClientOrgShort {
    id: string
    name: string
    image: string | undefined
    type: ClientOrgType
}
