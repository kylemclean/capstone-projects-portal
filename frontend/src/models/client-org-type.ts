enum ClientOrgType {
    Startup = "Startup",
    Nonprofit = "Non-profit",
    Academic = "Academic",
    CSL = "Community Service Learning",
    Other = "Other",
}

export default ClientOrgType
