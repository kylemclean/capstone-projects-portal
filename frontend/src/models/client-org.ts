import UserShort from "./user-short"
import ProjectShort from "./project-short"
import ClientOrgShort from "./client-org-short"

type ClientOrg = ClientOrgShort & {
    about: string
    website_link: string
    reps: UserShort[]
    projects: ProjectShort[]
    testimonial: string
}
export default ClientOrg
