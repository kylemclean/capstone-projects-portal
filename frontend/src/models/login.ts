import CurrentUserInfo from "./current-user-info"

export interface LoginWithEmailAndPasswordRequest {
    email: string
    password: string
}

export type OAuth2Provider = "GitHub"

export interface LoginWithOAuth2ProviderRequest {
    provider: OAuth2Provider
    code: string
}

export type LoginResult =
    | {
          success: true
          token: string
          user: CurrentUserInfo
      }
    | {
          success: false
          error: string
      }

export interface ActivateRequest {
    activationKey: string
    newPassword: string
}

export type ActivateResult = LoginResult

export type ResetPasswordRequest =
    | {
          currentPassword?: string
          newPassword: string
      }
    | {
          resetKey: string
          newPassword: string
      }

export type ResetPasswordResult = LoginResult

export interface RequestPasswordResetRequest {
    email: string
}
