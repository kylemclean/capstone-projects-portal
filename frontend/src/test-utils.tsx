import * as React from "react"
import { useLocation, useParams } from "react-router-dom"
import { screen, waitFor } from "@testing-library/react"
import { useCurrentUser } from "./hooks"

/**
 * Component to display the login status of the user
 */
export const LoginStatus = (): JSX.Element => {
    const currentUser = useCurrentUser()
    return (
        <div>
            {currentUser.logged_in
                ? `Logged in as ${currentUser.name}`
                : "Not logged in"}
        </div>
    )
}

/**
 * Mock home page that shows the login status
 */
export const MockHomePage = (): JSX.Element => (
    <div>
        Home page
        <LoginStatus />
    </div>
)

/**
 * Mock profile page that shows user ID
 */
export const MockProfilePage = (): JSX.Element => {
    const { id } = useParams<{ id: string }>()
    return <div>Profile page for {id}</div>
}

/**
 * Component that displays each query parameter in a list of "key=value" pairs
 * Requires the use of the `useLocation` hook
 */
export const QueryParamsDisplay = (): JSX.Element => {
    const { search } = useLocation()
    const params = new URLSearchParams(search)
    return (
        <div>
            <ul>
                {Array.from(params.entries()).map(([key, value]) => (
                    <li key={key}>
                        {key}={value}
                    </li>
                ))}
            </ul>
        </div>
    )
}

/**
 * Waits for a query parameter `name` to be set to `value`, or unset.
 * Requires a `QueryParamsDisplay` component to be rendered on the screen.
 * @param name the name of the query parameter
 * @param value the value of the query parameter, or `null` if it should not be present
 */
export const waitForQueryParam = async (
    name: string,
    value: string | null
): Promise<void> =>
    waitFor(() => {
        const text = `${name}=${value}`
        if (value !== null) screen.getByText(text)
        else expect(screen.queryByText(text)).toBeNull()
    })

/**
 * Mock login page that shows where it will redirect to after login
 */
export const MockLoginPage = (): JSX.Element => {
    const location = useLocation()
    const next = new URLSearchParams(location.search).get("next") ?? "/"
    return <div>Login page, will redirect to {next}</div>
}
