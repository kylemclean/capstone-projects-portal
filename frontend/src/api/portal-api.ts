import axios, {
    AxiosError,
    AxiosInstance,
    AxiosRequestConfig,
    AxiosResponse,
} from "axios"
import { Dispatch } from "react"
import ClientOrg from "../models/client-org"
import User from "../models/user"
import { axiosConfig } from "./config"
import Project from "../models/project"
import ProposalForm from "../models/proposal-form"
import {
    ActivateRequest,
    ActivateResult,
    LoginResult,
    LoginWithEmailAndPasswordRequest,
    LoginWithOAuth2ProviderRequest,
    RequestPasswordResetRequest,
    ResetPasswordRequest,
    ResetPasswordResult,
} from "../models/login"
import CurrentUserInfo from "../models/current-user-info"
import Action from "../global-state/action"
import ImportCsvResponse from "../models/import"

/**
 * Returns an appropriate error message for a given error object.
 * @param error The error object.
 * @returns The appropriate error message, or "Unknown error" if the error is of an unknown type or has no message.
 */
const errorMessageOf = (
    error: AxiosError<{ error: string }> | Error | unknown
): string =>
    (error as AxiosError<{ error: string }>)?.response?.data?.error ||
    (error as Error)?.message ||
    "Unknown error"

const onLoginResult = (result: LoginResult, dispatch: Dispatch<Action>) => {
    if (result.success === true) {
        localStorage.setItem("token", result.token)
        dispatch({ type: "SET_CURRENT_USER", value: result.user })
    }
}

const handleLoginResponsePromise = (
    promise: Promise<AxiosResponse<LoginResult>>,
    dispatch: Dispatch<Action>
): Promise<LoginResult> =>
    promise
        .then((response) => {
            onLoginResult(response.data, dispatch)
            return response.data
        })
        .catch((error) => {
            const result: LoginResult = {
                success: false,
                error: errorMessageOf(error),
            }
            onLoginResult(result, dispatch)
            return result
        })

export default class PortalApi {
    axiosInstance: AxiosInstance

    constructor(axiosRequestConfig: AxiosRequestConfig, logResponses: boolean) {
        // reusable AxiosInstance
        this.axiosInstance = axios.create(axiosRequestConfig)

        // Interceptor to include token in all requests if it is set
        this.axiosInstance.interceptors.request.use((requestConfig) => {
            const token = localStorage.getItem("token")
            if (token !== null)
                requestConfig.headers.Authorization = `Token ${token}`

            return requestConfig
        })

        // log all errors
        if (logResponses) {
            this.axiosInstance.interceptors.response.use(
                (response) =>
                    // Any status code that lie within the range of 2xx cause this function to trigger
                    // console.log(response)
                    response,
                (error) => {
                    // Any status codes that falls outside the range of 2xx cause this function to trigger
                    console.error(error)

                    // Invalidate token if it results in HTTP 401
                    if (
                        error.request?.status === 401 &&
                        localStorage.getItem("token") !== null
                    ) {
                        localStorage.removeItem("token")
                        // Reload page to remove stale information
                        window.location.reload()
                        return
                    }

                    throw error
                }
            )
        }
    }

    getClientOrgs = async (): Promise<ClientOrg[]> =>
        this.axiosInstance
            .get<ClientOrg[]>("/orgs/")
            .then((response) => response.data)

    getClientOrg = async (id: string): Promise<ClientOrg> =>
        this.axiosInstance
            .get<ClientOrg>(`/orgs/${id}/`)
            .then((response) => response.data)

    editClient = async (clientId: string, data: FormData): Promise<void> =>
        this.axiosInstance
            .patch<FormData>(`/orgs/${clientId}/`, data)
            .then(() => {})

    getUser = async (id: string): Promise<User> =>
        this.axiosInstance
            .get<User>(`/users/${id}/`)
            .then((response) => response.data)

    getImage = async (image_url: string | BlobPart): Promise<Response> =>
        this.axiosInstance
            .get<Response>(`${image_url}`)
            .then((response) => response)
            .catch((error) => error.response)

    getCurrentUserInfo = async (): Promise<CurrentUserInfo> =>
        this.axiosInstance
            .get<CurrentUserInfo>("/users/me/")
            .then((response) => response.data)

    getProjects = async (homePageOnly = false): Promise<Project[]> =>
        this.axiosInstance
            .get<Project[]>(`/projects/?home_page=${homePageOnly}`)
            .then((response) => response.data)

    getProject = async (id: string): Promise<Project> =>
        this.axiosInstance
            .get<Project>(`/projects/${id}/`)
            .then((response) => response.data)

    createProposal = async (inputData: ProposalForm): Promise<void> =>
        this.axiosInstance
            .post<ProposalForm>("/proposals/", inputData)
            .then(() => {})

    editProfile = async (userId: string, inputData: FormData): Promise<void> =>
        this.axiosInstance
            .patch<FormData>(`/users/${userId}/`, inputData)
            .then(() => {})

    editProject = async (
        projectId: string,
        projectData: FormData
    ): Promise<void> =>
        this.axiosInstance
            .patch<FormData>(`/projects/${projectId}/`, projectData)
            .then(() => {})

    validateCsv = async (csvFile: File): Promise<ImportCsvResponse> => {
        const formData = new FormData()
        formData.append("file", csvFile)
        return this.axiosInstance
            .post<FormData, AxiosResponse<ImportCsvResponse>>(
                "/csv/validate/",
                formData
            )
            .then((response) => response.data)
    }

    importCsv = async (csvFile: File): Promise<ImportCsvResponse> => {
        const formData = new FormData()
        formData.append("file", csvFile)
        return this.axiosInstance
            .post<FormData, AxiosResponse<ImportCsvResponse>>(
                "/csv/import/",
                formData
            )
            .then((response) => response.data)
    }

    loginWithEmailAndPassword = async (
        inputData: LoginWithEmailAndPasswordRequest,
        dispatch: Dispatch<Action>
    ): Promise<LoginResult> =>
        handleLoginResponsePromise(
            this.axiosInstance.post<
                LoginWithEmailAndPasswordRequest,
                AxiosResponse<LoginResult>
            >("/login/email/", inputData),
            dispatch
        )

    loginWithOAuth2Provider = async (
        inputData: LoginWithOAuth2ProviderRequest,
        dispatch: Dispatch<Action>
    ): Promise<LoginResult> =>
        handleLoginResponsePromise(
            this.axiosInstance.post<
                LoginWithOAuth2ProviderRequest,
                AxiosResponse<LoginResult>
            >("/login/oauth2/", inputData),
            dispatch
        )

    logout = async (dispatch: Dispatch<Action>): Promise<void> => {
        localStorage.removeItem("token")
        dispatch({ type: "SET_CURRENT_USER", value: { logged_in: false } })
    }

    activateAccount = async (
        inputData: ActivateRequest,
        dispatch: Dispatch<Action>
    ): Promise<ActivateResult> =>
        handleLoginResponsePromise(
            this.axiosInstance.post<
                ActivateRequest,
                AxiosResponse<ActivateResult>
            >("/activate/", inputData),
            dispatch
        )

    resetPassword = async (
        inputData: ResetPasswordRequest,
        dispatch: Dispatch<Action>
    ): Promise<ResetPasswordResult> =>
        handleLoginResponsePromise(
            this.axiosInstance.post<
                ResetPasswordRequest,
                AxiosResponse<ResetPasswordResult>
            >("/reset-password/", inputData),
            dispatch
        )

    requestPasswordReset = async (
        inputData: RequestPasswordResetRequest
    ): Promise<void> =>
        this.axiosInstance
            .post<RequestPasswordResetRequest>(
                "/request-password-reset/",
                inputData
            )
            .then(() => {})

    invalidateOtherSessions = async (
        dispatch: Dispatch<Action>
    ): Promise<LoginResult> =>
        handleLoginResponsePromise(
            this.axiosInstance.post<void, AxiosResponse<LoginResult>>(
                "/logout-all/"
            ),
            dispatch
        )

    logAllData = async (): Promise<void> => {
        console.log("%c Client Orgs:", "font-size: 16px")
        console.table(await this.getClientOrgs())
        console.log("%c Projects:", "font-size: 16px")
        console.table(await this.getProjects())
    }
}

const logResponses = false
export const portalApiInstance = new PortalApi(axiosConfig, logResponses)
