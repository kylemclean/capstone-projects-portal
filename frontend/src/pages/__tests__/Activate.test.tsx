import * as React from "react"
import { render, waitFor, screen, fireEvent } from "@testing-library/react"
import "@testing-library/jest-dom"
import { MemoryRouter, Route } from "react-router-dom"
import { http } from "msw"
import { setupServer } from "msw/node"
import GlobalStateProvider from "../../global-state/provider"
import ActivatePage from "../Activate"
import State from "../../global-state/state"
import { MockHomePage } from "../../test-utils"
import { ActivateRequest, LoginResult } from "../../models/login"
import { axiosConfig } from "../../api/config"

/**
 * Renders the Activate page in a router with a GlobalStateProvider.
 * @param key key to activate with
 * @param initialState initial state to set for the GlobalStateProvider. If undefined, the state will be the default.
 * @returns the return value of the render function
 */
const renderActivatePage = (key: string, initialState?: State) => {
    const url = `/activate/${key}`
    return render(
        <GlobalStateProvider initialState={initialState}>
            <MemoryRouter initialEntries={[url]}>
                <Route path="/activate/:key" exact component={ActivatePage} />
                <Route path="/" exact component={MockHomePage} />
            </MemoryRouter>
        </GlobalStateProvider>
    )
}

/**
 * @returns the password, confirm password, and submit button of the Activate page
 */
const getActivateFormControls = () => {
    const passwordInput = screen.getByLabelText("New password")
    const confirmPasswordInput = screen.getByLabelText("Confirm new password")
    const activateButton = screen.getByRole("button", { name: "Activate" })
    return { passwordInput, confirmPasswordInput, activateButton }
}

const VALID_KEY = "12345"
const INVALID_PASSWORD = "1"
const TOKEN = "HEREISYOURTOKEN"

const successResponse: LoginResult = {
    success: true,
    token: TOKEN,
    user: {
        logged_in: true,
        is_superuser: false,
        has_password: true,
        id: "1fc66da3-5f63-4dd8-aaac-425a18804cdc",
        name: "Guy User",
        image: "https://example.com/image.jpg",
    },
}
const invalidActivationKeyResponse: LoginResult = {
    success: false,
    error: "Invalid activation key",
}
const invalidNewPasswordResponse: LoginResult = {
    success: false,
    error: "Invalid new password",
}

const server = setupServer(
    // Mock /api/activate endpoint
    http.post(`${axiosConfig.baseURL}/activate/`, async ({ request }) => {
        const { activationKey, newPassword } =
            (await request.json()) as ActivateRequest

        if (activationKey !== VALID_KEY)
            return Response.json(invalidActivationKeyResponse, { status: 400 })
        if (newPassword === INVALID_PASSWORD)
            return Response.json(invalidNewPasswordResponse, { status: 400 })

        return Response.json(successResponse)
    })
)

beforeAll(() => server.listen({ onUnhandledRequest: "error" }))
afterEach(() => server.resetHandlers())

it("matches snapshot", async () => {
    const { container } = renderActivatePage(VALID_KEY)
    expect(container).toMatchSnapshot()
})

it("redirects to home page if logged in", async () => {
    renderActivatePage(VALID_KEY, {
        currentUser: {
            logged_in: true,
            is_superuser: false,
            has_password: true,
            id: "1fc66da3-5f63-4dd8-aaac-425a18804cdc",
            name: "Guy User",
            image: "https://example.com/image.jpg",
        },
    })

    // Assert that the user is redirected to the home page
    screen.getByText("Home page")
})

it("redirects to home page and logs the user in on successful activation", async () => {
    renderActivatePage(VALID_KEY)
    const { passwordInput, confirmPasswordInput, activateButton } =
        getActivateFormControls()

    // Fill in the form
    fireEvent.change(passwordInput, {
        target: { value: "thisismynewpassword" },
    })
    fireEvent.change(confirmPasswordInput, {
        target: { value: "thisismynewpassword" },
    })

    // Submit the form
    fireEvent.click(activateButton)

    // Assert that the activate button is disabled
    expect(activateButton).toBeDisabled()

    // Wait for the user to be redirected to the home page
    await waitFor(() => screen.getByText("Home page"))
})

it("shows an error if the activation key is invalid", async () => {
    renderActivatePage("invalidkey")
    const { passwordInput, confirmPasswordInput, activateButton } =
        getActivateFormControls()

    // Fill in the form
    fireEvent.change(passwordInput, {
        target: { value: "thisismynewpassword" },
    })
    fireEvent.change(confirmPasswordInput, {
        target: { value: "thisismynewpassword" },
    })

    // Submit the form
    fireEvent.click(activateButton)

    // Assert that the activate button is disabled
    expect(activateButton).toBeDisabled()

    // Wait for the error to be displayed
    await waitFor(() => screen.getByText("Invalid activation key"))

    // Assert that the activate button is reenabled
    expect(activateButton).toBeEnabled()
})

it("shows an error if the new password is invalid", async () => {
    renderActivatePage(VALID_KEY)
    const { passwordInput, confirmPasswordInput, activateButton } =
        getActivateFormControls()

    // Fill in the form
    fireEvent.change(passwordInput, { target: { value: INVALID_PASSWORD } })
    fireEvent.change(confirmPasswordInput, {
        target: { value: INVALID_PASSWORD },
    })

    // Submit the form
    fireEvent.click(activateButton)

    // Assert that the activate button is disabled
    expect(activateButton).toBeDisabled()

    // Wait for the error to be displayed
    await waitFor(() => screen.getByText("Invalid new password"))

    // Assert that the activate button is reenabled
    expect(activateButton).toBeEnabled()
})

it("shows an error if the new password and confirm password do not match", async () => {
    renderActivatePage(VALID_KEY)
    const { passwordInput, confirmPasswordInput, activateButton } =
        getActivateFormControls()

    // Fill in the form
    fireEvent.change(passwordInput, {
        target: { value: "thisismynewpassword" },
    })
    fireEvent.change(confirmPasswordInput, {
        target: { value: "thisisnotmynewpassword" },
    })

    // Submit the form
    fireEvent.click(activateButton)

    // Wait for the error to be displayed
    await waitFor(() => screen.getByText("Passwords must match"))

    // Assert that the confirm password input is marked as invalid and has an error message
    expect(confirmPasswordInput).toHaveAttribute("aria-invalid", "true")
    expect(confirmPasswordInput).toHaveAttribute(
        "aria-describedby",
        "confirmPassword-helper-text"
    )
})

it("shows an error if the password inputs empty", async () => {
    renderActivatePage(VALID_KEY)
    const { passwordInput, confirmPasswordInput, activateButton } =
        getActivateFormControls()

    // Submit the form
    fireEvent.click(activateButton)

    // Wait for the error to be displayed on the password input
    await waitFor(() => screen.getByText("New password is required"))

    // Assert that the confirm password input error is displayed as well
    screen.getByText("Confirm new password is required")

    // Assert that the password input is marked as invalid and has thane error message
    expect(passwordInput).toHaveAttribute("aria-invalid", "true")
    expect(passwordInput).toHaveAttribute(
        "aria-describedby",
        "newPassword-helper-text"
    )

    // Assert that the confirm password input is marked as invalid and has an error message
    expect(confirmPasswordInput).toHaveAttribute("aria-invalid", "true")
    expect(confirmPasswordInput).toHaveAttribute(
        "aria-describedby",
        "confirmPassword-helper-text"
    )
})

afterAll(() => server.close())
