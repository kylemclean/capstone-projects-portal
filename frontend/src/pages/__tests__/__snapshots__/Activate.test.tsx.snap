// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div>
  <div
    class="MuiContainer-root MuiContainer-maxWidthSm css-cuefkz-MuiContainer-root"
  >
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 css-f1hcus-MuiPaper-root"
    >
      <div
        class="MuiBox-root css-0"
      >
        <h1
          class="MuiTypography-root MuiTypography-h4 MuiTypography-alignCenter MuiTypography-gutterBottom css-1trk376-MuiTypography-root"
        >
          Activate Your Account
        </h1>
        <div
          class="MuiBox-root css-1xlzx9v"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
          >
            To activate your account, you must set a password.
          </p>
        </div>
        <form>
          <div
            class="MuiFormControl-root MuiFormControl-marginDense MuiFormControl-fullWidth MuiTextField-root css-1ldlpx-MuiFormControl-root-MuiTextField-root"
          >
            <label
              class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
              data-shrink="false"
              for="newPassword"
              id="newPassword-label"
            >
              New password
            </label>
            <div
              class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
            >
              <input
                aria-invalid="false"
                class="MuiOutlinedInput-input MuiInputBase-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                id="newPassword"
                name="newPassword"
                type="password"
                value=""
              />
              <fieldset
                aria-hidden="true"
                class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="css-dhh0px"
                >
                  <span>
                    New password
                  </span>
                </legend>
              </fieldset>
            </div>
            
          </div>
          <div
            class="MuiFormControl-root MuiFormControl-marginDense MuiFormControl-fullWidth MuiTextField-root css-1ldlpx-MuiFormControl-root-MuiTextField-root"
          >
            <label
              class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
              data-shrink="false"
              for="confirmPassword"
              id="confirmPassword-label"
            >
              Confirm new password
            </label>
            <div
              class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
            >
              <input
                aria-invalid="false"
                class="MuiOutlinedInput-input MuiInputBase-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                id="confirmPassword"
                name="confirmPassword"
                type="password"
                value=""
              />
              <fieldset
                aria-hidden="true"
                class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="css-dhh0px"
                >
                  <span>
                    Confirm new password
                  </span>
                </legend>
              </fieldset>
            </div>
            
          </div>
          <button
            class="MuiLoadingButton-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeLarge MuiButton-containedSizeLarge MuiButton-fullWidth MuiButtonBase-root css-1nelln8-MuiButtonBase-root-MuiButton-root-MuiLoadingButton-root"
            tabindex="0"
            type="submit"
          >
            Activate
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </form>
      </div>
    </div>
  </div>
</div>
`;
